beacon_node/http_server/src/lib.rs

```diff
@@ -0,0 +1,110 @@
+mod api;
+mod key;
+mod metrics;
+
+use beacon_chain::{BeaconChain, BeaconChainTypes};
+use futures::Future;
+use iron::prelude::*;
+use network::NetworkMessage;
+use router::Router;
+use slog::{info, o, warn};
+use std::sync::Arc;
+use tokio::runtime::TaskExecutor;
+
+#[derive(PartialEq, Clone, Debug)]
+pub struct HttpServerConfig {
+    pub enabled: bool,
+    pub listen_address: String,
+}
+
+impl Default for HttpServerConfig {
+    fn default() -> Self {
+        Self {
+            enabled: false,
+            listen_address: "127.0.0.1:5051".to_string(),
+        }
+    }
+}
+
+/// Build the `iron` HTTP server, defining the core routes.
+pub fn create_iron_http_server<T: BeaconChainTypes + 'static>(
+    beacon_chain: Arc<BeaconChain<T>>,
+) -> Iron<Router> {
+    let mut router = Router::new();
+
+    // A `GET` request to `/metrics` is handled by the `metrics` module.
+    router.get(
+        "/metrics",
+        metrics::build_handler(beacon_chain.clone()),
+        "metrics",
+    );
+
+    // Any request to all other endpoints is handled by the `api` module.
+    router.any("/*", api::build_handler(beacon_chain.clone()), "api");
```

Comment:
Another way to separate these is to "mount" the API handler on `/api/`, and the metrics handler on `/metrics/` using the [`Mount` middleware](https://docs.rs/mount/0.4.0/mount/struct.Mount.html). Then we could have URLs like `/api/node/fork`, etc. Probably not worth changing for now, but would be worth thinking about once we have more APIs and we want to stabilise them